### PR TITLE
changed output finding + add error to catch problem in future

### PR DESCRIPTION
**Pull request by sadolini**
Root cause analysis:
Loader creates graph with Split node for parallelcomponent. 2 issues happens:
* we add Identity to this node as node without output, because it was not included in used_layers (really it has outputs and Identity is not needed)
* as result we have Split with 2 expected outputs and 3 real outputs - this situation incorrectly processed by set_ports transformation and split wrong branches. As result, shapes were calculated incorrectly and partial infer fails.

Solution: 
For the first issue, remove logic with adding layer to used_layers to find out layers without outputs because nnet1 model is chain and only last layer can be without output. 
To avoid creating wrong graph for similar situations, check on number real outputs and expected is added. If output_count is 1, then we can have several outputs from 1 port. If we have more branches (output_count>1), all of them should be on different ports, otherwise we can't detect in loader which branches on 1 port and which on different. 

Ticket: 51842

Code:
* [x]  Comments
* [x]  Code style (PEP8)
* [x]  Transformation generates reshape-able IR **Not applicable**
* [x]  Transformation preserves original framework node names **Not applicable**

Validation:
* [x]  Unit tests
* [x]  Framework operation tests **Not applicable**
* [x]  Transformation tests **Not applicable**
* [x]  Model Optimizer IR Reader check **Not applicable**

Documentation:
* [x]  Supported frameworks operations list **Not applicable**
* [x]  Guide on how to convert the **public** model **Not applicable**
* [x]  User guide update **Not applicable**

### CODE REVIEW

**Review by vgavrilo — APPROVED** [review]

**Review by popovaan — APPROVED** [review]

File: model-optimizer/extensions/front/kaldi/set_ports.py [hunk 1]
```diff
@@ -28,6 +29,11 @@ def find_and_replace_pattern(self, graph: Graph):
 
             in_ports_count = node.in_ports_count if node.has_valid('in_ports_count') else len(inputs)
             out_ports_count = node.out_ports_count if node.has_valid('out_ports_count') else len(outputs)
+
+            if len(outputs) > out_ports_count > 1:
+                raise Error("Node have more children then it should: " +
+                            "should be {} but there is {}".format(out_ports_count, len(outputs)))
```
Comment (rkazants) [inline model-optimizer/extensions/front/kaldi/set_ports.py:35]: please correct the sentence and add a node name: "Node {} has more outputs than expected ones: {} - expected and {} -actual".format(...)

File: model-optimizer/mo/front/kaldi/loader/loader.py [hunk 2]
```diff
@@ -158,17 +159,16 @@ def load_kalid_nnet1_model(graph, file_descr, name):
...
         log.debug('{} (type is {}) was loaded'.format(prev_layer_id, component_type))
 
     # Tensor names information corresponding to a node is stored on outgoing edges.
     # As output nodes do not have outgoing edges, fake outputs are required. In the following code
     # for each output Identity node is added, and tensor name for the output is kept
     # on (output, fake output) edge. After Result nodes adding transformation fake outputs
     # are deleted from graph.
-    output_layers = graph.nodes - used_layers
-    add_outputs_identity(graph, output_layers, lambda g, output, fake_output: g.create_edge(
+    add_outputs_identity(graph, [output_layer], lambda g, output, fake_output: g.create_edge(
```
Comment (rkazants) [inline model-optimizer/mo/front/kaldi/loader/loader.py:171]: should we insert assert here if `output_layer is None`?

File: model-optimizer/extensions/front/kaldi/set_ports.py (hunk 1 above)
Comment (sadolini) [reply model-optimizer/extensions/front/kaldi/set_ports.py]: fixed

File: model-optimizer/extensions/front/kaldi/set_ports.py [hunk 3]
```diff
@@ -31,8 +31,8 @@ def find_and_replace_pattern(self, graph: Graph):
             out_ports_count = node.out_ports_count if node.has_valid('out_ports_count') else len(outputs)
 
             if len(outputs) > out_ports_count > 1:
-                raise Error("Node have more children then it should: " +
-                            "should be {} but there is {}".format(out_ports_count, len(outputs)))
+                raise Error("Node {} have more children then it should: " +
```
Comment (rkazants) [inline model-optimizer/extensions/front/kaldi/set_ports.py:34]: "Node have" -> "Node has"

File: model-optimizer/mo/front/kaldi/loader/loader.py [hunk 4]
```diff
@@ -168,6 +168,7 @@ def load_kalid_nnet1_model(graph, file_descr, name):
     # for each output Identity node is added, and tensor name for the output is kept
     # on (output, fake output) edge. After Result nodes adding transformation fake outputs
     # are deleted from graph.
+    assert output_layer is not None, "Output layer not found in graph"
```
Comment (rkazants) [inline model-optimizer/mo/front/kaldi/loader/loader.py:171]: "Output layer not found in graph" -> "Output layer is not found in graph"

File: model-optimizer/extensions/front/kaldi/set_ports.py [hunk 5]
```diff
@@ -28,6 +29,11 @@ def find_and_replace_pattern(self, graph: Graph):
 
             in_ports_count = node.in_ports_count if node.has_valid('in_ports_count') else len(inputs)
             out_ports_count = node.out_ports_count if node.has_valid('out_ports_count') else len(outputs)
+
+            if len(outputs) > out_ports_count > 1:
+                raise Error("Node {} has more children then it should: " +
+                            "should be {} but there is {}".format(node_id, out_ports_count, len(outputs)))
```
Comment (rkazants) [inline model-optimizer/extensions/front/kaldi/set_ports.py:35]: then -> than.
Please double-check the sentence in terms of grammar correctness.

**Review by rkazants — APPROVED** [review]

**Review by lazarevevgeny — APPROVED** [review]

**Pull request merged**